plugins/woocommerce/src/Blocks/BlockTypes/ProductStockIndicator.php

```diff
@@ -78,42 +56,46 @@ protected static function getTextBasedOnStock( $is_in_stock, $is_low_stock, $low
 	 * @return string Rendered block type output.
 	 */
 	protected function render( $attributes, $content, $block ) {
-		if ( ! empty( $content ) ) {
-			parent::register_block_type_assets();
-			$this->register_chunk_translations( [ $this->block_name ] );
-			return $content;
-		}
-
 		$post_id = isset( $block->context['postId'] ) ? $block->context['postId'] : '';
 		$product = wc_get_product( $post_id );
 
-		if ( ! $product ) {
+		if ( ! $product || in_array( $product->get_type(), $this->product_types_without_stock_indicator(), true ) ) {
 			return '';
 		}
 
-		$is_in_stock     = $product->is_in_stock();
-		$is_on_backorder = $product->is_on_backorder();
+		$availability = $product->get_availability();
 
-		$low_stock_amount = $product->get_low_stock_amount();
-		$total_stock      = $product->get_stock_quantity();
-		$is_low_stock     = $low_stock_amount && $total_stock <= $low_stock_amount;
+		if ( empty( $availability['availability'] ) ) {
+			return '';
+		}
 
+		$low_stock_amount   = $product->get_low_stock_amount();
+		$total_stock        = $product->get_stock_quantity();
+		$is_low_stock       = $low_stock_amount && $total_stock <= $low_stock_amount;
 		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
 
 		$classnames  = isset( $classes_and_styles['classes'] ) ? ' ' . $classes_and_styles['classes'] . ' ' : '';
-		$classnames .= ! $is_in_stock ? ' wc-block-components-product-stock-indicator--out-of-stock ' : '';
-		$classnames .= $is_in_stock ? ' wc-block-components-product-stock-indicator--in-stock ' : '';
-		$classnames .= $is_low_stock ? ' wc-block-components-product-stock-indicator--low-stock ' : '';
-		$classnames .= $is_on_backorder ? ' wc-block-components-product-stock-indicator--available-on-backorder ' : '';
+		$classnames .= sprintf( ' wc-block-components-product-stock-indicator--%s', $availability['class'] );
+
+		$is_backorder_notification_visible = $product->is_in_stock() && $product->backorders_require_notification();
+
+		if ( empty( $content ) && $is_backorder_notification_visible && $total_stock > 0 ) {
```

Comment:
Yes, it's a little confusing, but it's possible to have `$product->is_in_stock()` as `true` and  `$total_stock > 0` as `false`, because we are not managing stock.

Reply:
Oh, ok. Makes sense. Thanks for explaining it!